examples/restaurant_visits/run_without_frameworks_utility_analysis.py

```diff
@@ -0,0 +1,103 @@
+# Copyright 2022 OpenMined.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+""" Demo of running PipelineDP locally, without any external data processing framework.
+
+This demo outputs a utility analysis of errors and noise for each partition in the dataset.
+
+1. Install Python and run on the command line `pip install pipeline-dp absl-py`
+2. Run python python run_without_frameworks_utility_analysis.py --input_file=<path to data.txt from 3> --output_file=<...>
+"""
+
+from absl import app
+from absl import flags
+import numpy as np
+import pipeline_dp
+import pandas as pd
+from utility_analysis_new.dp_engine import UtilityAnalysisEngine
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string('input_file', 'restaurants_week_data.csv',
+                    'The file with the restaurant visits data')
+flags.DEFINE_string('output_file', None, 'Output file')
+
+
+def write_to_file(col, filename):
+    with open(filename, 'w') as out:
+        out.write('\n'.join(map(str, col)))
+
+
+def main(unused_argv):
+    # Here, we use a local backend for computations. This does not depend on
+    # any pipeline framework and it is implemented in pure Python in
+    # PipelineDP. It keeps all data in memory and is not optimized for large data.
+    # For datasets smaller than ~tens of megabytes, local execution without any
+    # framework is faster than local mode with Beam or Spark.
+    backend = pipeline_dp.LocalBackend()
+
+    # Define the privacy budget available for our computation.
+    budget_accountant = pipeline_dp.NaiveBudgetAccountant(total_epsilon=1,
+                                                          total_delta=1e-6)
+
+    # Load and parse input data
+    df = pd.read_csv(FLAGS.input_file)
+    df.rename(inplace=True,
+              columns={
+                  'VisitorId': 'user_id',
+                  'Time entered': 'enter_time',
+                  'Time spent (minutes)': 'spent_minutes',
+                  'Money spent (euros)': 'spent_money',
+                  'Day': 'day'
+              })
+    # Double the inputs so we have twice as many contributions per partition
+    df_double = pd.DataFrame(np.repeat(df.values, 2, axis=0))
```

Comment:
Would 
`df_double = pd.concat([df, df])` 
work?

Reply:
Yes, this works well so we don't need to bring in the numpy dependency! Thank you!